### PR TITLE
feat(wallet): Dune stats

### PR DESCRIPTION
**Pull request by DiogoSoaress**
## What it solves
Fetches Safe numbers from Dune queries

I ended up moving the fetchers outside of the hook.

### CODE REVIEW

File: src/hooks/useSafeStats.ts [hunk 1]
```diff
@@ -31,12 +34,25 @@ export const fetchTotalSafesDeployed = async (): Promise<number | null> => {
     .catch(() => null)
 }
 
+export const fetchMonthlyActiveUsers = async (): Promise<number | null> => {
+  return fetch(duneQueryUrlBuilder(QUERY_ID_MONTHLY_ACTIVE_USERS, DUNE_API_KEY))
+    .then((res) => res.json())
```
Comment (iamacook) [inline src/hooks/useSafeStats.ts:39]: The `res` may be erroneous. We should check that `res.ok` before returning the JSON.

File: src/hooks/useSafeStats.ts [hunk 2]
```diff
@@ -31,12 +34,25 @@ export const fetchTotalSafesDeployed = async (): Promise<number | null> => {
...
   const formattedTotalSafesDeployed = totalSafesDeployed ? formatValue(totalSafesDeployed) : null
+  const formattedMonthlyActiveUsers = monthlyActiveUsers ? formatValue(monthlyActiveUsers) : null
 
-  return [formattedTotalTransactions, formattedTotalAssets, formattedTotalSafesDeployed]
+  return [
+    formattedTotalTransactions,
+    formattedTotalBalanceUsd,
+    formattedTotalSafesDeployed,
+    formattedMonthlyActiveUsers,
+  ]
```
Comment (iamacook) [inline src/hooks/useSafeStats.ts:57]: Nit: it's easy to access the wrong index. What do you think about returning an object instead?

File: src/pages/wallet.tsx [hunk 3]
```diff
@@ -2,15 +2,26 @@ import type { InferGetStaticPropsType, NextPage } from 'next'
...
 
 const FAQ_CONTENT_TYPE_ID = '1jCIVFDUzFO1okK8b6TTxS'
 
 const WalletPage: NextPage<InferGetStaticPropsType<typeof getStaticProps>> = (props) => <Wallet {...props} />
 
 export async function getStaticProps() {
   const faqContent = await client.getEntry<TypeBaseBlockSkeleton>(FAQ_CONTENT_TYPE_ID)
+  const dataRoomStats = await fetchDataRoomStats()
+  const totalBalanceUsd = await fetchTotalBalanceUsd()
+  const monthlyActiveUsers = await fetchMonthlyActiveUsers()
```
Comment (iamacook) [inline src/pages/wallet.tsx:16]: We could make these a single promise with `Promise.all`.

File: src/hooks/useSafeStats.ts [hunk 4]
```diff
@@ -31,12 +34,25 @@ export const fetchTotalSafesDeployed = async (): Promise<number | null> => {
...
 }
 
+export const fetchMonthlyActiveUsers = async (): Promise<number | null> => {
+  return fetch(duneQueryUrlBuilder(QUERY_ID_MONTHLY_ACTIVE_USERS, DUNE_API_KEY))
+    .then((res) => res.json())
+    .then((data) => data.result.rows[data.result.rows.length - 1].active_users)
+    .catch(() => null)
+}
+
 export const useSafeStats = (): Array<string | null> => {
```
Comment (iamacook) [inline src/hooks/useSafeStats.ts:44]: Have you thought about using `swr` for this hook? We'd not need a context.

File: src/hooks/useSafeStats.ts (hunk 2 above)
Comment (DiogoSoaress) [reply src/hooks/useSafeStats.ts]: Followed the recommendation 👍

File: src/hooks/useSafeStats.ts (hunk 1 above)
Comment (DiogoSoaress) [reply src/hooks/useSafeStats.ts]: Added to the other fetchers as well.

**Pull request merged**